include/swift/SIL/MemoryLocations.h

```diff
@@ -247,22 +255,38 @@ class MemoryLocations {
 
   /// Clears the location bits of \p addr in \p bits, if \p addr is associated
   /// with a location.
-  void clearBits(Bits &bits, SILValue addr) const {
-    if (auto *loc = getLocation(addr))
-      bits.reset(loc->subLocations);
+  void clearBits(Bits &bits, SILValue addr, bool evenTrivial = false) {
+    if (auto *loc = getLocation(addr)) {
+      const auto &sublocations = [&]() {
+        if (!preserveTrivialLocations || evenTrivial) {
+          return loc->subLocations;
+        } else {
+          return loc->subLocations & getNonTrivialLocations();
```

Comment:
Maybe it would be better to push this into bitwise anding into the MemoryVerifier instead of adding this mode to MemoryLocations..

Reply:
I think it doesn't matter. MemoryLocations are only used in the MemoryLifetimeVerifier and DestroyHoisting. DestroyHoisting is not enabled anymore and we can delete it. So The MemoryLifetimeVerifier will be the only user of MemoryLocations.

Reply:
Removed this new flag `preserveTrivialLocations`, the old flag `handleTrivialLocations` and `DestroyHoisting`.